src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IInterpolatedStringExpression.cs

```diff
@@ -0,0 +1,295 @@
+        [Fact, WorkItem(18300, "https://github.com/dotnet/roslyn/issues/18300")]
+        public void InterpolatedStringExpression_EmptyInterpolationPart()
+        {
+            string source = @"
+using System;
+
+internal class Class
+{
+    public void M()
+    {
+        Console.WriteLine(/*<bind>*/$""{}""/*</bind>*/);
+    }
+}
+";
+            string expectedOperationTree = @"
+IInterpolatedStringExpression (OperationKind.InterpolatedStringExpression, Type: System.String, IsInvalid) (Syntax: '$""{}""')
+  Parts(1): IInterpolation (OperationKind.Interpolation, IsInvalid) (Syntax: '{}')
+      Expression: IInvalidExpression (OperationKind.InvalidExpression, Type: ?, IsInvalid) (Syntax: '')
+";
+            var expectedDiagnostics = new DiagnosticDescription[] {
+                // CS1733: Expected expression
+                //         Console.WriteLine(/*<bind>*/$"{}"/*</bind>*/);
+                Diagnostic(ErrorCode.ERR_ExpressionExpected, "").WithLocation(8, 40)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<InterpolatedStringExpressionSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [Fact, WorkItem(18300, "https://github.com/dotnet/roslyn/issues/18300")]
+        public void InterpolatedStringExpression_TextAndInterpolationParts()
+        {
+            string source = @"
+using System;
+
+internal class Class
+{
+    public void M(int x)
+    {
+        Console.WriteLine(/*<bind>*/$""String {x} and constant {1}""/*</bind>*/);
+    }
+}
+";
+            string expectedOperationTree = @"
+IInterpolatedStringExpression (OperationKind.InterpolatedStringExpression, Type: System.String) (Syntax: '$""String {x ... nstant {1}""')
+  Parts(4): IInterpolatedStringText (OperationKind.InterpolatedStringText) (Syntax: 'String ')
+      Text: ILiteralExpression (Text: String ) (OperationKind.LiteralExpression, Type: System.String, Constant: ""String "") (Syntax: 'String ')
+    IInterpolation (OperationKind.Interpolation) (Syntax: '{x}')
+      Expression: IParameterReferenceExpression: x (OperationKind.ParameterReferenceExpression, Type: System.Int32) (Syntax: 'x')
+    IInterpolatedStringText (OperationKind.InterpolatedStringText) (Syntax: ' and constant ')
+      Text: ILiteralExpression (Text:  and constant ) (OperationKind.LiteralExpression, Type: System.String, Constant: "" and constant "") (Syntax: ' and constant ')
+    IInterpolation (OperationKind.Interpolation) (Syntax: '{1}')
+      Expression: ILiteralExpression (Text: 1) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1) (Syntax: '1')
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+
+            VerifyOperationTreeAndDiagnosticsForTest<InterpolatedStringExpressionSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [Fact, WorkItem(18300, "https://github.com/dotnet/roslyn/issues/18300")]
+        public void InterpolatedStringExpression_FormatAndAlignment()
```

Comment:
>InterpolatedStringExpression_FormatAndAlignment() [](start = 20, length = 49)

It doesn't look like we have a test for a scenario when all three members of IInterpolation are present.